File: server/src/com/twilightimperium/backend/data/SystemData.java
package com.twilightimperium.backend.data;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;


public class SystemData {


  public static final Map<String, SystemModel> systemList = new HashMap<>();

  static {
    systemList.put("Mecatol Rex", new SystemModel(new ArrayList<>(List.of(PlanetData.planets.get("Mecatol Rex")))));
    systemList.put("Abyz", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Abyz"),
        PlanetData.planets.get("Fria")
    ))));
    systemList.put("Arinam", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Arinam"),
        PlanetData.planets.get("Meer")
    ))));
    systemList.put("Arnor", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Arnor"),
        PlanetData.planets.get("Lor")
    ))));
    systemList.put("Bereg", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Bereg"),
        PlanetData.planets.get("Lirta IV")
    ))));
    systemList.put("Centauri", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Centauri"),
        PlanetData.planets.get("Gral")
    ))));
    systemList.put("Corneeq", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Corneeq"),
        PlanetData.planets.get("Resculon")
    ))));
    systemList.put("Dal Bootha", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Dal Bootha"),
        PlanetData.planets.get("Xxehan")
    ))));
    systemList.put("Lazar", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Lazar"),
        PlanetData.planets.get("Sakulag")
    ))));
    systemList.put("Lodor", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Lodor")
    )), SystemModel.Wormhole.alpha));
    systemList.put("Mehar Xull", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Mehar Xull")
    ))));
    systemList.put("Mellon", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Mellon"),
        PlanetData.planets.get("Zohbat")
    ))));
    systemList.put("New Albion", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("New Albion"),
        PlanetData.planets.get("Starpoint")
    ))));
    systemList.put("Quann", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Quann")
    )), SystemModel.Wormhole.beta));
    systemList.put("Qucen'n", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Qucen'n"),
        PlanetData.planets.get("Rarron")
    ))));
    systemList.put("Saudor", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Saudor")
    ))));
    systemList.put("Tar'Mann", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Tar'Mann")
    ))));
    systemList.put("Tequ'ran", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Tequ'ran"),
        PlanetData.planets.get("Torkan")
    ))));
    systemList.put("Thibah", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Thibah")
    ))));
    systemList.put("Vefut II", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Vefut II")
    ))));
    systemList.put("Wellon", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Wellon")
    ))));
    // ---------------------------------- HOME SYSTEMS ----------------------------------
    systemList.put("Nestphar", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Nestphar")
    )), "arborec"));
    systemList.put("Arc Prime", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Arc Prime"),
        PlanetData.planets.get("Wren Terra")
    )), "letnev"));
    systemList.put("Lisis II", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Lisis II"),
        PlanetData.planets.get("Ragh")
    )), "saar"));
    systemList.put("Muaat", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Muaat")
    )), "muaat"));
    systemList.put("Hercant", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Hercant"),
        PlanetData.planets.get("Arretze"),
        PlanetData.planets.get("Kamdorn")
    )), "hacan"));
    systemList.put("Jord", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Jord")
    )), "sol"));
    systemList.put("Creuss", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Creuss")
    )), SystemModel.Wormhole.delta, "creuss"));
    systemList.put("[0.0.0]", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("[0.0.0]")
    )), "l1z1x"));
    systemList.put("Moll Primus", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Moll Primus")
    )), "mentak"));
    systemList.put("Druaa", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Druaa"),
        PlanetData.planets.get("Maaluuk")
    )), "naalu"));
    systemList.put("Mordai II", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Mordai II")
    )), "nekro"));
    systemList.put("Tren'Lak", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Tren'Lak"),
        PlanetData.planets.get("Quinarra")
    )), "sardakk"));
    systemList.put("Jol", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Jol"),
        PlanetData.planets.get("Nar")
    )), "jol_nar"));
    systemList.put("Winnu", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Winnu")
    )), "winnu"));
    systemList.put("Archon Ren", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Archon Ren"),
        PlanetData.planets.get("Archon Tau")
    )), "xxcha"));
    systemList.put("Darien", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Darien")
    )), "yin"));
    systemList.put("Retillion", new SystemModel(new ArrayList<>(List.of(
        PlanetData.planets.get("Retillion"),
        PlanetData.planets.get("Shalloq")
    )), "yssaril"));
    systemList.put("Asteroid", new SystemModel(SystemModel.Anomaly.asteroid));
    systemList.put("Supernova", new SystemModel(SystemModel.Anomaly.asteroid));
    systemList.put("Rift", new SystemModel(SystemModel.Anomaly.asteroid));
    systemList.put("Nebula", new SystemModel(SystemModel.Anomaly.asteroid));
    systemList.put("WormholeAlpha", new SystemModel(SystemModel.Wormhole.alpha));
    systemList.put("WormholeBeta", new SystemModel(SystemModel.Wormhole.beta));
    systemList.put("CreussGate", new SystemModel(SystemModel.Wormhole.delta));
    systemList.put("Empty", new SystemModel());
    systemList.put("Undefined", new SystemModel());
}
}

File: client/lib/data/system_data.dart
import 'package:client/data/planet_data.dart';

//This differs from the SystemState, it only includes the basic info of a system
class SystemModel {
  List<PlanetModel>? planets;
  Anomaly? anomaly;
  Wormhole? wormhole;
  String? homeSystem;

  SystemModel({this.planets, this.anomaly, this.wormhole, this.homeSystem});
}

enum Wormhole { alpha, beta, delta, gamma }

enum Anomaly { asteroid, nebula, supernova, rift }

class SystemData {
  static final Map<String, SystemModel> systemList = {
    "Mecatol Rex": SystemModel(
      planets: [PlanetData.planets["Mecatol Rex"]!],
    ),
    "Abyz": SystemModel(
      planets: [
        PlanetData.planets["Abyz"]!,
        PlanetData.planets["Fria"]!,
      ],
    ),
    "Arinam": SystemModel(
      planets: [
        PlanetData.planets["Arinam"]!,
        PlanetData.planets["Meer"]!,
      ],
    ),
    "Arnor": SystemModel(
      planets: [
        PlanetData.planets["Arnor"]!,
        PlanetData.planets["Lor"]!,
      ],
    ),
    "Bereg": SystemModel(
      planets: [
        PlanetData.planets["Bereg"]!,
        PlanetData.planets["Lirta IV"]!,
      ],
    ),
    "Centauri": SystemModel(
      planets: [
        PlanetData.planets["Centauri"]!,
        PlanetData.planets["Gral"]!,
      ],
    ),
    "Corneeq": SystemModel(
      planets: [
        PlanetData.planets["Corneeq"]!,
        PlanetData.planets["Resculon"]!,
      ],
    ),
    "Dal Bootha": SystemModel(
      planets: [
        PlanetData.planets["Dal Bootha"]!,
        PlanetData.planets["Xxehan"]!,
      ],
    ),
    "Lazar": SystemModel(
      planets: [
        PlanetData.planets["Lazar"]!,
        PlanetData.planets["Sakulag"]!,
      ],
    ),
    "Lodor": SystemModel(
      planets: [
        PlanetData.planets["Lodor"]!,
      ],
      wormhole: Wormhole.alpha,
    ),
    "Mehar Xull": SystemModel(
      planets: [
        PlanetData.planets["Mehar Xull"]!,
      ],
    ),
    "Mellon": SystemModel(
      planets: [
        PlanetData.planets["Mellon"]!,
        PlanetData.planets["Zohbat"]!,
      ],
    ),
    "New Albion": SystemModel(
      planets: [
        PlanetData.planets["New Albion"]!,
        PlanetData.planets["Starpoint"]!,
      ],
    ),
    "Quann": SystemModel(
      planets: [
        PlanetData.planets["Quann"]!,
      ],
      wormhole: Wormhole.beta,
    ),
    "Qucen'n": SystemModel(
      planets: [
        PlanetData.planets["Qucen'n"]!,
        PlanetData.planets["Rarron"]!,
      ],
    ),
    "Saudor": SystemModel(
      planets: [
        PlanetData.planets["Saudor"]!,
      ],
    ),
    "Tar'Mann": SystemModel(
      planets: [
        PlanetData.planets["Tar'Mann"]!,
      ],
    ),
    "Tequ'ran": SystemModel(
      planets: [
        PlanetData.planets["Tequ'ran"]!,
        PlanetData.planets["Torkan"]!,
      ],
    ),
    "Thibah": SystemModel(
      planets: [
        PlanetData.planets["Thibah"]!,
      ],
    ),
    "Vefut II": SystemModel(
      planets: [
        PlanetData.planets["Vefut II"]!,
      ],
    ),
    "Wellon": SystemModel(
      planets: [
        PlanetData.planets["Wellon"]!,
      ],
    ),
    "Nestphar": SystemModel(
      planets: [
        PlanetData.planets["Nestphar"]!,
      ],
      homeSystem: "Arborec",
    ),
    "Arc Prime": SystemModel(
      planets: [
        PlanetData.planets["Arc Prime"]!,
        PlanetData.planets["Wren Terra"]!,
      ],
      homeSystem: "Letnev",
    ),
    "Lisis II": SystemModel(
      planets: [
        PlanetData.planets["Lisis II"]!,
        PlanetData.planets["Ragh"]!,
      ],
      homeSystem: "Saar",
    ),
    "Muaat": SystemModel(
      planets: [
        PlanetData.planets["Muaat"]!,
      ],
      homeSystem: "Muaat",
    ),
    "Hercant": SystemModel(
      planets: [
        PlanetData.planets["Hercant"]!,
        PlanetData.planets["Arretze"]!,
        PlanetData.planets["Kamdorn"]!,
      ],
      homeSystem: "Hacan",
    ),
    "Jord": SystemModel(
      planets: [
        PlanetData.planets["Jord"]!,
      ],
      homeSystem: "Sol",
    ),
    "Creuss": SystemModel(
      planets: [
        PlanetData.planets["Creuss"]!,
      ],
      wormhole: Wormhole.delta,
      homeSystem: "Creuss",
    ),
    "[0.0.0]": SystemModel(
      planets: [
        PlanetData.planets["[0.0.0]"]!,
      ],
      homeSystem: "L1Z1X",
    ),
    "Moll Primus": SystemModel(
      planets: [
        PlanetData.planets["Moll Primus"]!,
      ],
      homeSystem: "Mentak",
    ),
    "Druaa": SystemModel(
      planets: [
        PlanetData.planets["Druaa"]!,
        PlanetData.planets["Maaluuk"]!,
      ],
      homeSystem: "Naalu",
    ),
    "Mordai II": SystemModel(
      planets: [
        PlanetData.planets["Mordai II"]!,
      ],
      homeSystem: "Nekro",
    ),
    "Tren'Lak": SystemModel(
      planets: [
        PlanetData.planets["Tren'Lak"]!,
        PlanetData.planets["Quinarra"]!,
      ],
      homeSystem: "Sardakk",
    ),
    "Jol": SystemModel(
      planets: [
        PlanetData.planets["Jol"]!,
        PlanetData.planets["Nar"]!,
      ],
      homeSystem: "Jol-Nar",
    ),
    "Winnu": SystemModel(
      planets: [
        PlanetData.planets["Winnu"]!,
      ],
      homeSystem: "Winnu",
    ),
    "Archon Ren": SystemModel(
      planets: [
        PlanetData.planets["Archon Ren"]!,
        PlanetData.planets["Archon Tau"]!,
      ],
      homeSystem: "Xxcha",
    ),
    "Darien": SystemModel(
      planets: [
        PlanetData.planets["Darien"]!,
      ],
      homeSystem: "Yin",
    ),
    "Retillion": SystemModel(
      planets: [
        PlanetData.planets["Retillion"]!,
        PlanetData.planets["Shalloq"]!,
      ],
      homeSystem: "Yssaril",
    ),
    "Asteroid": SystemModel(
      anomaly: Anomaly.asteroid,
    ),
    "Supernova": SystemModel(
      anomaly: Anomaly.supernova,
    ),
    "Rift": SystemModel(
      anomaly: Anomaly.rift,
    ),
    "Nebula": SystemModel(
      anomaly: Anomaly.nebula,
    ),
    "WormholeAlpha": SystemModel(
      wormhole: Wormhole.alpha,
    ),
    "WormholeBeta": SystemModel(
      wormhole: Wormhole.beta,
    ),
    "CreussGate": SystemModel(
      wormhole: Wormhole.delta,
    ),
    "Empty": SystemModel(),
    "Undefined": SystemModel()
  };
}
